velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -417,6 +417,41 @@ TEST_F(ArithmeticTest, isNanDouble) {
   EXPECT_EQ(false, isNan(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, hexWithBigint) {
+  const auto toHex = [&](std::optional<int64_t> value) {
+    return evaluateOnce<std::string>("hex(c0)", value);
+  };
+  EXPECT_EQ("11", toHex(17));
+  EXPECT_EQ("FFFFFFFFFFFFFFEF", toHex(-17));
+  EXPECT_EQ("0", toHex(0));
+  EXPECT_EQ("FFFFFFFFFFFFFFFF", toHex(-1));
+  EXPECT_EQ("7FFFFFFFFFFFFFFF", toHex(INT64_MAX));
+  EXPECT_EQ("8000000000000000", toHex(INT64_MIN));
+}
+
+TEST_F(ArithmeticTest, hexWithVarbinaryAndVarchar) {
+  const auto toHex = [&](std::optional<std::string> value) {
+    return std::pair(
+        evaluateOnce<std::string>("hex(cast(c0 as varbinary))", value),
+        evaluateOnce<std::string>("hex(c0)", value));
+  };
+  auto result = toHex("");
+  EXPECT_TRUE(result.first == result.second && result.first == "");
```

Comment:
split this into 2 checks

EXPECT_EQ(result.first, result.second)
EXPECT_EQ(result.first, "")

Actually, better to modify toHex to assert the results are the same and return these results.

```
const auto toHex = [&](std::optional<std::string> value) {
    auto varbinaryResult =
        evaluateOnce<std::string>("hex(cast(c0 as varbinary))", value);
    auto varcharResult = evaluateOnce<std::string>("hex(c0)", value));
   
    EXPECT_TRUE(varcharResult.hasValue());
    EXPECT_TRUE(varbinaryResult.hasValue());
    EXPECT_EQ(varcharResult.value(), varbinaryResult.value());

    return varcharResult.value();
  };
```

ASSER_EQ(toHex("Spark SQL"), "537061726B2053514C");